docs/dev/developer-guides/transactions/blocks.md

```diff
@@ -59,10 +64,36 @@ Block hashes in zkSync are deterministic and are derived from the following form
+
+In Solidity, `block.number` and `block.timestamp` returns the number and timestamp of the L1 batch respectively.
+
+### zksync-web3 SDK
+
+The following are the block properties returned when you use the `getBlock` method via the **zksync-web3** API.
+
+| Parameter     | Description                                                                                                             |
+| ------------- | ------------------------------------------------------------------------------------------------------------------------|
+| hash          | The hash of the block.                                                                                                  |
+| parentHash    | It refers to the hash of the parent block in L2.                                                                        |
+| number        | the number of the current block                                                                                         |
+| timestamp     | The current block's creation time in seconds.                                                                           |
+| nonce         | It's the most recent transaction based on the account's counter, which maintains track of how many transactions it does.|
+| difficulty    | The current block difficulty returns 2500000000000000 (zkSync does not have proof of work consensus).                   |
+| gasLimit      | The block gas limit, always returns `2^32-1`.                                                                           |
+| gasUsed       | The actual amount of gas used in this block.                                                                            |
+| transactions  | A list of all transactions included in the block.                                                                       |
+| baseFeePerGas | The EIP1559-like baseFee for this block.                                                                                |
+
+### Why do we return L2 blocks from API?
+
+On zkSync Era we return L2 blocks from API because this is how all platforms, which include SDKs, Metamask and all other popular wallets can perceive our transactions as processed. It is expected that a transaction is processed once it is included in some block. That's why we need to produce L2 blocks faster than L1 batches.
+
+### Why do we return L1 batches inside EVM?
 
-- Timestamp: The current block's creation time in seconds that returns the timestamp of the L1 batch.
-- Block number: The unique sequential number for this block.
-- Gas limit: The current block gas limit, always returns `2^32-1`.
-- Coinbase: The current block miner’s address, returns the [bootloader](../system-contracts.md#bootloader) address.
-- Difficulty: The current block difficulty, returns `2500000000000000` (zkSync does not have proof of work consensus).
+In the past,`block.number` was a part of the VM and was provided once per batch and so we returned the number of the batch there.
```

Comment:
For me personally it is confusing compination of words: *`block.number` was a part of the VM.*..and then *Presently, we return the number of L1 batch inside the VM*
if @StanislavBreadless  is ok with this I do not have a strong opinion

Reply:
Yeah, probably better to rephrase it, though not sure exactly how. Maybe just not mention how it used to be? We've been using a system contract and not a VM env param for`block.number` for a few months already, I don't think anybody cares how it used to be at this point